pdfannots/printer/markdown.py

```diff
@@ -286,29 +288,58 @@ def emit_body(
 
         self._fmt_header_called = False
 
-        def fmt_header(name: str) -> str:
+        def fmt_header(name: str, level: int = 2) -> str:
+            """
+            A function that formats a header with a given name and level.
+
+            Parameters:
+                name (str): The name of the header.
+                level (int, optional): The level of the header. Defaults to 2.
+
+            Returns:
+                str: The formatted header.
+            """
             # emit blank separator line if needed
             prefix = '\n' if self._fmt_header_called else ''
             self._fmt_header_called = True
-            return prefix + "## " + name + "\n\n"
+            header = '#' * level
+            return prefix + header + " " + name + "\n"
 
         # Partition annotations into nits, comments, and highlights.
         nits = []
         comments = []
+        highlights_by_color: typ.Dict[str, typ.List[Annotation]] = {}
         highlights = []
         for a in document.iter_annots():
             if a.subtype in self.ANNOT_NITS:
                 nits.append(a)
             elif a.contents:
                 comments.append(a)
             elif a.subtype == AnnotationType.Highlight:
-                highlights.append(a)
+                if self.group_highlights_by_color:
+                    if a.color:
+                        color = str(a.color.ashex())
+                    else:
+                        color = "undefined"
+                    if color not in highlights_by_color:
```

Comment:
I won't insist on it, but you can use a defaultdict(list) to avoid this check -- https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
Switched to a defaultdict in newest commit and saved two lines.